sdk/keyvault/keyvault-keys/src/core/challengeBasedAuthenticationPolicy.ts

```diff
@@ -49,34 +76,42 @@ export class ChallengeBasedAuthenticationPolicy extends BaseRequestPolicy {
     nextPolicy: RequestPolicy,
     options: RequestPolicyOptions,
     private credential: TokenCredential,
-    private tokenCache: AccessTokenCache
+    private tokenCache: AccessTokenCache,
+    private challengeCache: AuthenticationChallengeCache
   ) {
     super(nextPolicy, options);
   }
 
-  private parseWWWAuthenticate(www_authenticate: string): string {
+  private parseWWWAuthenticate(www_authenticate: string): {
+    authorization: string,
+    resource: string
+  } {
+    const returnValue = {
+      authorization: "",
+      resource: ""
+    };
     // Parses an authentication message like:
     // ```
     // Bearer authorization="some_authorization", resource="https://some.url"
     // ```
-    let authenticateArray = www_authenticate.split(" ");
-
-    // Remove the "Bearer" piece
-    delete authenticateArray[0];
+    let spaceSep = www_authenticate.split(" ");
 
     // Split the KV comma-separated list
-    let commaSep = authenticateArray.join().split(",");
-    for (let item of commaSep) {
-      // Split the key/value pairs
-      let kv = item.split("=");
-      if (kv[0].trim() == "resource") {
-        // Remove the quotations around the string
-        let resource = kv[1].trim().replace(/['"]+/g, '');
-
-        return resource;
+    for (const spaceItem of spaceSep) {
```

Comment:
Is there a good place to centralize this so we don't repeat the logic?

Reply:
I can move this challenge based authentication to a common folder. Should I do this as part of the PR? I'm not sure how big the change would be, but it might not be that big (it will be net negative for sure).

Reply:
Here's a draft PR moving this to a common folder, but it's not really working. I will ask the team for ideas tomorrow: https://github.com/sadasant/azure-sdk-for-js/pull/19

Reply:
I made this issue to track this work https://github.com/Azure/azure-sdk-for-js/issues/8634 I don't think this issue should block this PR though.